pytest/tests/loadtest/locust/common/base.py

```diff
@@ -115,8 +130,100 @@ def sign_and_serialize(self, block_hash) -> bytes:
             sender.key, sub.account_id, sub, int(self.balance * 1E24),
             sender.use_nonce(), block_hash)
 
+    def sender_id(self) -> str:
+        return self.sender.key.account_id
+
+
+class NearNodeProxy:
+    """
+    Wrapper around a RPC node connection that tracks requests on locust.
+    """
+
+    def __init__(self, host, request_event):
+        self.request_event = request_event
+        url, port = host.split(":")
+        self.node = cluster.RpcNode(url, port)
+
+    def send_tx(self, tx: Transaction, locust_name):
+        """
+        Send a transaction and return the result, no retry attempted.
+        """
+        block_hash = base58.b58decode(self.node.get_latest_block().hash)
+        signed_tx = tx.sign_and_serialize(block_hash)
+
+        # doesn't work because it raises on status
+        # rpc_result = self.node.send_tx_and_wait(...)
 
-class NearUser(HttpUser):
+        j = {
```

Comment:
yeah I refactored it into a function like this:

```python
    def post_json(self, method: str, params: list(str)):
        j = {
            "method": method,
            "params": params,
            "id": "dontcare",
            "jsonrpc": "2.0"
        }
        return self.session.post(url="http://%s:%s" % self.node.rpc_addr(),
                                 json=j)
```

This also includes session reuse. (I ran into some problems because `requests.post` opens a new HTTP connection on every request, whereas the locust HttpUser reused them.)